Show tab hover state only on hovered split tab with SideBySide

simonhong opened the pull request:
fix https://github.com/brave/brave-browser/issues/47126

Chromium set hovered state when any tab in split tabs is hovered.
But we only want hover effect only on hovered tab in split tab.

TEST=SideBySideEnabledBrowserTest.SelectTabTest

Manual test:
1. Open Brave with `--enable-features=SideBySide`
2. Open split tabs
3. Hover on tab in split tab
4. Check only that hovered tab has hover effect

<!-- Add brave-browser issue below that this PR will resolve -->
Resolves

<!-- CI-related labels that can be applied to this PR:
* CI/run-audit-deps (1) - check for known npm/cargo vulnerabilities (audit_deps)
* CI/run-network-audit (1) - run network-audit
* CI/run-upstream-tests - run Chromium unit and browser tests on Linux and Windows (otherwise only on Linux)
* CI/run-linux-arm64, CI/run-macos-x64, CI/run-windows-arm64, CI/run-windows-x86 - run builds that would otherwise be skipped
* CI/skip - do not run CI builds (except noplatform)
* CI/skip-linux-x64, CI/skip-android, CI/skip-macos-arm64, CI/skip-ios, CI/skip-windows-x64 - skip CI builds for specific platforms
* CI/skip-upstream-tests - do not run Chromium unit, or browser tests (otherwise only on Linux)
* CI/skip-all-linters - do not run presubmit and lint checks
* CI/storybook-url (1) - deploy storybook and provide a unique URL for each build

(1) applied automatically when some files are changed (see: https://github.com/brave/brave-core/blob/master/.github/labeler.yml)
-->

<!--
## Checklist:

- Review design docs
  [Browser design principles](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/chrome_browser_design_principles.md)
  [Style guide](https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++.md)
  [Core principles](https://www.chromium.org/developers/core-principles/)
- Ensure there are (tests)[https://www.chromium.org/developers/testing/]. Unit test as much as possible (including edge cases), but also include browser tests covering high level functionality.
- Ensure that there are comments explaining what classes/methods are/do. The "why" is often more important than the "what" in comments. Also update any relevant docs (moving docs from wiki to brave-core if necessary).
- Request security or other review (third-party libraries, rust code, etc...) if applicable [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) [other review](https://github.com/brave/reviews/issues/new/choose)
  Also see [adding third-party libraries](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/adding_to_third_party.md) for general guidelines on using third party code
- Maks sure there is a [ticket](https://github.com/brave/brave-browser/issues) for your issue
- Use Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- Write a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- Squash any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- Add appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- Run `git rebase master` (if needed)
-->

brave-builds commented:
> [!WARNING]
> You have got a presubmit warning. Please address it if possible.

```
A banned function was used.
    browser/ui/views/split_view/split_view_browsertest.cc:63:
      Do not RunUntilIdle. If possible, explicitly quit the run loop using run_loop.Quit() or run_loop.QuitClosure() if completion can be observed using a lambda or callback. Otherwise, wait for the condition to be true via base::test::RunUntil().
```

<!-- presubmit-hash=8c12c5017839b14f568be92b7673ebbbcd709d257060aeb87593bb1e2774e543 -->

brave-builds commented:
> [!WARNING]
> You have got a presubmit warning. Please address it if possible.

```
A banned function was used.
    browser/ui/views/split_view/split_view_browsertest.cc:66:
      Do not RunUntilIdle. If possible, explicitly quit the run loop using run_loop.Quit() or run_loop.QuitClosure() if completion can be observed using a lambda or callback. Otherwise, wait for the condition to be true via base::test::RunUntil().
```

<!-- presubmit-hash=b5766f0c6b01b0c294ed69595d2b630efd49e368d15ea14407f914ac89724f68 -->

simonhong commented:
> This PR got warning about `RunUntilIdle()`. Curious it's warning or must be followed? I think I couldn't test this PR w/o using `RunUntilIdle()` as current test checks event that will not be happened.

NVM! Updated test code and deleted `RunUntilIdle()`.

sangwoo108 reviewed (approved):
LGTM 👍
